pandas/tslib.pyx

```diff
@@ -4822,8 +4833,23 @@ def dates_normalized(ndarray[int64_t] stamps, tz=None):
 #----------------------------------------------------------------------
 
 def isleapyear(int64_t year):
+    msg = "isleapyear is deprecated. Use .is_leap_year property instead"
+    warnings.warn(msg, FutureWarning)
     return is_leapyear(year)
 
+
+cpdef isleapyear_arr(ndarray years):
```

Comment:
Should we make this private? (``_isleapyear_arr``) Because if we regard functions here as internal, we actually don't need to deprecate isleapyear and could just remove it